### PR TITLE
Only refund quest scroll to leader if quest was active

### PR DESCRIPTION
**Pull request by phillipthelen**
otherwise quest leader gets more quest scrolls than they had to begin with.

### CODE REVIEW

File: src/controllers/groups.js [hunk 1]
```diff
@@ -607,11 +607,11 @@ api.questAbort = function(req, res, next){
     },
     // Refund party leader quest scroll
     function(cb){
-      var update = {$inc:{}};
-      update['$inc']['items.quests.' + group.quest.key] = 1;
-      User.update({_id:group.quest.leader}, update, cb);
-    },
-    function(cb) {
```
Comment (lefnire) [inline src/controllers/groups.js:614]: This PR combines the last two functions into one (with the above two deletions). `async.parallel` will then determine this function complete on `User.update(..., cb)` and ignore the rest of the code, and we'll lose the `done` function's error-handler for `group.save()`. 

Un-delete the above two lines if you could & push, then I'll merge

**Comment by phillipthelen** [comment]
Oh. I didn't know that async.parallel would determine it complete at that point.
I merged the two functions on purpose.
1. If they are run in parallel I wasn't sure if group.quest.active still was set or it if it already was cleared. (maybe I'm wrong here and it's okay for them to run in parallel? I don't really have much JS experience)
2. When the if statement in the first function evaluated to false, async.parallel aparently does not know, that the function is complete and would hang forever. Is there a manual way to tell async.parallel, that the execution is completed?

**Comment by lefnire** [comment]
Oh, I see! Ok, In that case I'd say just remove `cb` from `User.save()`, since the done function doesn't work-with/return `user`, but _does_ with `group`. Ie, deem the function complete after it's operated on `group` so we can return a fully-modified `group` object (in it's current state, it may return `res.json(group)` before `group.quest = {key:null,progress:{},leader:null};`.  Ie, change to this:

``` javascript
User.update({_id:group.quest.leader}, update);
```

Pretty minor though, compared to what I thought was happening before. Thanks for the explanation!

**Comment by lefnire** [comment]
Good catch btw @vIiRuS, I didn't realize this was actually fixing 2 bugs

**Comment by phillipthelen** [comment]
Ok, hope I got everything right now :) sorry for the mishap.

**Comment by lefnire** [comment]
`contrib.lvl++`, thanks!

**Pull request merged**